portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,28 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  List<String> messages = new ArrayList<>();
```

Comment:
Move this variable definition into the `doGet` method. If you define it at the class level, then every time someone refreshes your page, your code will add another 3 comments to the array. Eventually your server will OOM after so many visits.

Reply:
Did this change get made?

Reply:
It seems doGet always returns the same messages, this could be made a constant ImmutableList instead.

Reply:
Oh oops I don't think it got changed, I'll quickly move the variable definition.